apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -43,12 +103,16 @@ Services.obs.addObserver(function(document) {
         storagePromise = appWindow.TestStorages.getStorage('messagesDB');
       }
 
-      storagePromise = storagePromise.then(function(storage) {
-        return storage || {
-          threads: new Map(),
-          recipientToThreadId: new Map(),
-          uniqueMessageIdCounter: 0
-        };
+      storagePromise = storagePromise.then((storage) => {
+        if (!storage) {
+          return {
+            threads: new Map(),
+            recipientToThreadId: new Map(),
+            uniqueMessageIdCounter: 0
+          };
+        }
+
+        return rehydrateAttachments(storage).then(() => storage);
       });
```

Comment:
Regarding rejection if no storage found - good idea, why not, I'll consider that in one of the future integration related patches (just don't want to wait for Treeherder once again :p) :)